VIC-749-rs-profile

patric-dosch-vi opened the pull request.

patric-dosch-vi committed:
feat: VIC-749 RS profile | Berater sees demographic data & topic of the RS Profil

- demographics data and topic data should be in the ConsultantSessionDTO

patric-dosch-vi committed:
feat: VIC-749 RS profile | Berater sees demographic data & topic of the RS Profil

- demographics data and topic data should be in the ConsultantSessionDTO
- minor SonarLint changes

patric-dosch-vi commented:
Hi @tkuzynow, yesterday the PR didn't work out because I had to restructure the tests. Unfortunately this popped up only yesterday, strange.

I have now decided to use a second TopicEnricher. 1) I didn't find any interface impl for openAPI (there is some kind of inheritance https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/ in openAPI, but this is also not given in the generated classes) 2) the Session has only one topic, the ConsultantSession has mainTopic and topics.
You could still align mainTopic and topic. However, I find mainTopic much better than topic.

The only thing I would change is, to add a Mapper for "convertToSessionTopicDTO", but time is running :)
Best, Patric

tkuzynow commented on src/main/java/de/caritas/cob/userservice/api/service/session/ConsultantSessionTopicEnrichmentService.java:
```diff
@@ -0,0 +1,85 @@
+  private final @NonNull TopicService topicService;
+
+  public ConsultantSessionDTO enrichSessionWithTopicData(ConsultantSessionDTO session) {
+    if (session.getMainTopic() != null && session.getMainTopic().getId() != null
```
according to clean code, please extract the content of the if condition to a separate method like shouldEnrichTopics

tkuzynow commented on src/main/java/de/caritas/cob/userservice/api/service/session/ConsultantSessionTopicEnrichmentService.java:
```diff
@@ -0,0 +1,85 @@
+  public ConsultantSessionDTO enrichSessionWithTopicData(ConsultantSessionDTO session) {
+    if (session.getMainTopic() != null && session.getMainTopic().getId() != null
+        || session.getTopics() != null && !session.getTopics().isEmpty()) {
+      var availableTopics = topicService.getAllTopicsMap();
```
personally I think it would be a bit cleaner to have separate method to enrich mainTopic (also checking that this enrichment is needed in if statement) and a method to enrich all topics (also checking that this enrichment is needed in if statement). The reason for that is that only for digi we will have topics collection filled.

Otherwise you also need to amend log.debug message in else., as this is not accurate:
 "Skipping topic enrichment, topic id is not set for session with id: {}",
          session.getId());

patric-dosch-vi replied:
I hope I understood it as you ment

tkuzynow commented on src/test/java/de/caritas/cob/userservice/api/service/session/SessionServiceTest.java:
```diff
@@ -590,6 +592,48 @@ void fetchSessionForConsultant_Should_Return_ValidConsultantSessionDTO() {
     assertEquals(session.getGroupId(), result.getGroupId());
     assertEquals(session.getFeedbackGroupId(), result.getFeedbackGroupId());
     assertEquals(session.getConsultingTypeId(), result.getConsultingType().intValue());
+    assertEquals(session.getUserAge(), result.getAge());
```
if it's not too much effort, could you please change this test to use assertJ assertThat fluent assertions instead?

patric-dosch-vi replied:
hmm no no much, but I don't like that mix :/ also in the same class. So I think a bit more, that you should keep consitency in the place around.

tkuzynow replied:
agreed, it should be changed for the whole class or kept consitent as it is. So if changing for the whole class is not an option for the moment, it can stay as it is

patric-dosch-vi replied:
I agrree

patric-dosch-vi committed:
feat: VIC-749 RS profile | Berater sees demographic data & topic of the RS Profil

- ConsultantSessionTopicEnrichmentService have to separate methods for enriching mainTopic and topics
- changing for PR comments
